internal/impl/ollama/moderation_processor.go

```diff
@@ -0,0 +1,169 @@
+
+For more information, see the https://github.com/ollama/ollama/tree/main/docs[Ollama documentation^].`).
+		Version("4.41.0").
+		Fields(
+			service.NewStringAnnotatedEnumField(bopFieldModel, map[string]string{
+				"llama-guard3": "When using llama-guard3, two pieces of metadata is added: @safe with the value of `yes` or `no` and the second being @category for the safety category violation. For more information see the https://ollama.com/library/llama-guard3[Llama Guard 3 Model Card].",
+				"shieldgemma":  "When using shieldgemma, the model output is a single piece of metadata of @safe with a value of `yes` or `no` if the response is not in violation of its defined safety policies.",
+			}).
+				Description("The name of the Ollama LLM to use.").
+				Examples("llama-guard3", "shieldgemma"),
+			service.NewInterpolatedStringField(ompFieldUserPrompt).
+				Description("The input prompt that was used with the LLM. If using `ollama_chat` the you can use `save_prompt_metadata` to safe the prompt as metadata."),
+			service.NewInterpolatedStringField(ompFieldAssistantResponse).
+				Description("The LLM's response to classify if it contains safe or unsafe content."),
+		).Fields(commonFields()...).
+		Example(
+			"Use Llama Guard 3 classify a LLM response",
+			"This example uses Llama Guard 3 to check if another model responded with a safe or unsafe content.",
+			`
+input:
+  stdin:
+    scanner:
+      lines: {}
+pipeline:
+  processors:
+    - ollama_chat:
+        model: llava
+        prompt: "${!content().string()}"
+        save_prompt_metadata: true
+    - ollama_moderation:
+        model: llama-guard3
+        prompt: "${!@prompt}"
+        response: "${!content().string()}"
+    - mapping: |
+        root.response = content().string()
+        root.is_safe = @safe
+output:
+  stdout:
+    codec: lines
+`)
+}
+
+func makeOllamaModerationProcessor(conf *service.ParsedConfig, mgr *service.Resources) (service.Processor, error) {
+	p := ollamaModerationProcessor{}
+	var err error
+	p.prompt, err = conf.FieldInterpolatedString(ompFieldUserPrompt)
+	if err != nil {
+		return nil, err
+	}
+	p.response, err = conf.FieldInterpolatedString(ompFieldAssistantResponse)
+	if err != nil {
+		return nil, err
+	}
+	b, err := newBaseProcessor(conf, mgr)
+	if err != nil {
+		return nil, err
+	}
+	p.baseOllamaProcessor = b
+	return &p, nil
+}
+
+type ollamaModerationProcessor struct {
+	*baseOllamaProcessor
+
+	prompt   *service.InterpolatedString
+	response *service.InterpolatedString
+}
+
+func (o *ollamaModerationProcessor) Process(ctx context.Context, msg *service.Message) (service.MessageBatch, error) {
+	p, err := o.prompt.TryString(msg)
+	if err != nil {
+		return nil, err
+	}
+	r, err := o.response.TryString(msg)
+	if err != nil {
+		return nil, err
+	}
+	g, err := o.generateCompletion(ctx, p, r)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Might be worth adding some extra context to these errors.

Reply:
Done!